### DISCUSSION TITLE
thanhle.blog/nhung-loi-ngu-ngok-khi-lam-state-o-frontend

### DISCUSSION THREAD

**Discussion by giscus** [thanhlmm/blog#56] [Ideas]
# thanhle.blog/nhung-loi-ngu-ngok-khi-lam-state-o-frontend

Các lỗi thường gặp khi làm state ở Frontend mà mọi người nên tránh

https://thanhle.blog/blog/nhung-loi-ngu-ngok-khi-lam-state-o-frontend

**Comment by duongductrong** [comment]
Bài viết bác rất là đỉnh và cuống :v sắp cày hết đống article của bác r =)))). But có lên tay k thì chắc phải practice nhiều

**Reply by thanhlmm** [reply]
Đó là case anh hơi lo lắng. Đọc thì hay đó nhưng lúc làm lại chưa thấm được 😭

**Comment by nguyenhoaikhang37** [comment]
Đọc qua một loạt series về state trên blog anh em có cái nhìn hoàn toàn mới về việc đặt state luôn, trước h cứ đặt vô tội vạ, đọc hoài nó thấm, h mỗi lần tạo state mới đều nhớ tới 3 câu hỏi thần thánh kia, bài viết kĩ thuật trên blog của anh lúc nào cũng chất lượng

**Comment by Kennn-dev** [comment]
thực sự thì bài này dễ hiểu hơn 2 bài về state trước của bác x4 lần lun 👨‍🦳👨‍🦳

**Reply by thanhlmm** [reply]
Mình đọc lại thấy chối chối nên mới viết thêm mấy bài sau á. Mà là nó dễ hiểu hơn thay vì khó hiểu hơn :))))))

**Comment by thanhlmm** [comment]
Some nice articles:
- https://react-query-v3.tanstack.com/guides/does-this-replace-client-state
- https://react-query-v3.tanstack.com/comparison

**Comment by quangnmwork** [comment]
Well , một số lí do mà mình nghĩ khônng nên cứ vứt state lên global state 
1. Codebase khó quản lý , vì đôi khi nếu cứ vứt lên một số global state (redux ,...) khá là overkill
2. Không thích hợp dùng cho manage state components
Nếu có lí do nào khác vui lòng góp ý vs mình

**Reply by thanhlmm** [reply]
Codebase khó quản lý , vì đôi khi nếu cứ vứt lên một số global state (redux ,...) khá là overkill

Ý 1 mình thấy không đúng cũng không sai, bạn chọn làm việc nào và trade-off là gì thôi. Việc vứt tất cả state ra global thì giống mình vứt bất cứ file nào ra desktop vậy, nó vẫn chạy chỉ là càng ngày nó sẽ càng rối. Lúc đầu ít file thì sẽ gọi là dễ quản lý, lúc sau nhiều file thì sẽ gọi là... thằng này bừa bộn vkl. Nên viết cần có có chiến lược code để ở đâu, như thế nào là đang trade-off giữa thuận tiện của hiện tại và tương lai

Ý 2 thì mình ko hiểu bạn muốn truyền đạt gì

**Reply by quangnmwork** [reply]
Ý 2 của mình là ví dụ modal  component . Không cần thiết phải sử dụng redux hay thư viện quản lý state cho lắm (maybe i'm wrong)

**Reply by thanhlmm** [reply]
À z mình hoàn toàn đồng ý. Local state là đủ rồi

**Comment by TrungLeThanh** [comment]
<3

**Comment by thanhlmm** [comment]
**Redux is frequently used but generally disliked**. Redux is a state container for JavaScript apps, which helps centralize the application state for complex projects. Surprisingly, more survey respondents said they used and disliked it (47%) vs those who used and liked it (37%).

The reason for these mixed feelings comes from Redux having a much higher barrier to entry in learning how to use it well. Also, building complex applications without proper state management is unwise to do. Redux offers predictable state management. However, in return, it demands you understand its opinionated approach to this problem space, using the View/Actions/State model and updating state in an immutable way.

Redux brings unfamiliar concepts to many frontend engineers, such as Action Creators, Reducers, Store, Dispatch and Selectors. While all of this might sound overly complex, complex mobile applications tend to follow the model that Redux enforces: making state changes in immutable ways. Learn more about Redux here.

Nguồn: https://newsletter.pragmaticengineer.com/p/state-of-frontend-2022
